### PR TITLE
✨clusterctl: list images

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
This PR adds the the `--list-images` flag to `clusterctl init` so the user can list of the images required for completing at init operation.

**Which issue(s) this PR fixes**:
Fixes: https://github.com/kubernetes-sigs/cluster-api/issues/1829
Rif #1729 

/area clusterctl
/assign @vincepri 
/assing @ncdc 
/assing @wfernandes

### CODE REVIEW

File: cmd/clusterctl/pkg/client/cluster/cert_manager.go [hunk 1]
```diff
@@ -161,6 +193,24 @@ func (cm *certManagerClient) getWebHook(c client.Client) (*unstructured.Unstruct
 	return webHook, nil
 }
 
+// hasWebHook returns true if there is already a web-hook in the cluster
+func (cm *certManagerClient) hasWebHook() (bool, error) {
```
Comment (vincepri) [inline cmd/clusterctl/pkg/client/cluster/cert_manager.go]:
```suggestion
func (cm *certManagerClient) hasWebhook() (bool, error) {
```

Comment (vincepri) [reply cmd/clusterctl/pkg/client/cluster/cert_manager.go]: Very minor

Comment (fabriziopandini) [reply cmd/clusterctl/pkg/client/cluster/cert_manager.go]: The same change applies to many other places, so I'm sending a separated PR with s/WebHook/Webhook

**Comment by fabriziopandini** [comment]
@vincepri @wfernandes precedences are merged, removing WIP

File: cmd/clusterctl/pkg/client/client.go [hunk 2]
```diff
@@ -101,6 +101,9 @@ type Client interface {
 	// Init initializes a management cluster by adding the requested list of providers.
 	Init(options InitOptions) ([]Components, error)
 
+	// Init returns the list of images required for executing the init command.
```
Comment (wfernandes) [inline cmd/clusterctl/pkg/client/client.go]: nit:
```suggestion
	// InitImages returns the list of images required for executing the init command.
```

File: cmd/clusterctl/pkg/client/init.go [hunk 3]
```diff
@@ -130,6 +95,93 @@ func (c *clusterctlClient) Init(options InitOptions) ([]Components, error) {
 	return aliasComponents, nil
 }
 
+// Init returns the list of images required for init.
```
Comment (wfernandes) [inline cmd/clusterctl/pkg/client/init.go]: nit: `InitImages`

Comment (vincepri) [reply cmd/clusterctl/pkg/client/init.go]: well, seems like we need a follow-up PR :)

Comment (vincepri) [reply cmd/clusterctl/pkg/client/init.go]: I added a commit with your other suggestion

**Comment by wfernandes** [comment]
woops...my bad. I didn't even see this get merged in. I was just going through it.

**Pull request merged**